mobile/packages/swap/adapters/api/minswap/transformers.ts

```diff
@@ -0,0 +1,364 @@
+      },
+      true,
+    )
+  }
+
+  // Transform Minswap paths to Swap.Split format
+  const transformPathsToSplits = (paths: Array<any>): Swap.Split[] => {
+    if (!paths || paths.length === 0) return []
+
+    return paths
+      .map((path) => {
+        // Each path is an array of hops
+        const firstHop = path[0]
+        if (!firstHop) return null
+
+        const split = freeze(
+          {
+            amountIn: parseFloat(firstHop.amount_in || '0'),
+            batcherFee: 0, // Minswap doesn't provide batcher fee per split
+            deposits: parseFloat(firstHop.deposits || '0'),
+            protocol: mapDexToProtocol(firstHop.protocol),
+            expectedOutput: parseFloat(firstHop.amount_out || '0'),
+            expectedOutputWithoutSlippage: parseFloat(
+              firstHop.min_amount_out || '0',
+            ),
+            fee: parseFloat(firstHop.dex_fee || '0'),
+            initialPrice:
+              parseFloat(firstHop.amount_out || '0') /
+              parseFloat(firstHop.amount_in || '1'),
+            finalPrice:
+              parseFloat(firstHop.amount_out || '0') /
+              parseFloat(firstHop.amount_in || '1'),
+            poolFee: parseFloat(firstHop.lp_fee || '0'),
+            poolId: firstHop.pool_id || '',
+            priceDistortion: 0, // Minswap doesn't provide this
+            priceImpact: firstHop.price_impact || 0,
+          },
+          true,
+        )
+        return split
+      })
+      .filter(Boolean) as Swap.Split[]
+  }
+
+  return freeze(
+    {
+      tokens: {
+        request: (): TokensRequest => ({
+          query: '', // Empty string returns all tokens
+          only_verified: false, // Return all tokens, not just verified ones
+        }),
+        response: (data: TokensResponse): Portfolio.Token.Info[] => {
+          // Handle both flat structure (actual API) and nested structure (expected by types)
+          const transformedTokens = data.tokens.map((token) => {
+            // If token has 'asset' property, use it (nested structure)
+            // Otherwise, use the token directly (flat structure)
+            const tokenData = 'asset' in token ? (token as any).asset : token
+            return transformToken(tokenData)
+          })
+
+          return transformedTokens.filter(isNonNullable)
+        },
+      },
+
+      orders: {
+        response: (data: PendingOrdersResponse): Swap.Order[] => {
+          return data.orders
+            .filter(
+              (order) => order.token_in?.token_id && order.token_out?.token_id,
+            )
+            .map((order) =>
+              freeze(
+                {
+                  aggregator: Swap.Aggregator.Minswap,
+                  protocol: Swap.Protocol.Minswap_v2,
+                  placedAt: order.created_at,
+                  lastUpdate: order.created_at,
+                  status: 'open' as const,
+                  tokenIn: order.token_in.token_id as `${string}.${string}`,
```

Comment:
doesnt it need to be transformed? or they return in the `policyId.assetNameHex` mode? if so it should cast to `Portfolio.Token.Id` instead of string string